hls4ml/templates/vivado_accelerator_config.py

```diff
@@ -0,0 +1,143 @@
+        assert len(
+            model_outputs) == 1, "Only models with one output tensor are currently supported by VivadoAcceleratorBackend"
+        self.inp = model_inputs[0]
+        self.out = model_outputs[0]
+        inp_axi_t = self.input_type
+        out_axi_t = self.output_type
+
+        if inp_axi_t not in ['float', 'double']:
+            self.input_type = self._next_factor8_type(config.backend.convert_precision_string(inp_axi_t))
+        if out_axi_t not in ['float', 'double']:
+            self.output_type = self._next_factor8_type(config.backend.convert_precision_string(out_axi_t))
+
+        if self.input_type is 'float':
+            self.input_bitwidth = 32
+        elif self.input_type is 'double':
+            self.input_bitwidth = 64
+        else:
+            self.input_bitwidth = inp_axi_t.width
+
+        if out_axi_t is 'float':
+            self.output_bitwidth = 32
+        elif out_axi_t is 'double':
+            self.output_bitwidth = 64
+        else:
+            self.output_bitwidth = out_axi_t.width
+
+    def _next_factor8_type(self, p):
+        ''' Return a new type with the width rounded to the next factor of 8 up to p's width
+            Args:
+                p : IntegerPrecisionType or FixedPrecisionType
+            Returns:
+                An IntegerPrecisionType or FixedPrecisionType with the width rounder up to the next factor of 8
+                of p's width. Other parameters (fractional bits, extra modes) stay the same.
+        '''
+        W = p.width
+        newW = int(np.ceil(W / 8) * 8)
+        if isinstance(p, FixedPrecisionType):
+            return FixedPrecisionType(newW, p.integer, p.signed, p.rounding_mode, p.saturation_mode,
+                                      p.saturation_bits)
+        elif isinstance(p, IntegerPrecisionType):
+            return IntegerPrecisionType(newW, p.signed)
+
+    def get_io_bitwidth(self):
+        return self.input_bitwidth, self.output_bitwidth
+
+    def get_corrected_types(self):
+        return self.input_type, self.output_type, self.inp, self.out
+
+    def get_interface(self):
+        return self.interface
+
+    def get_board_info(self, board=None):
+        boards = get_supported_boards_dict()
+        if board is None:
+            board = self.board
+        if board in boards.keys():
+            return boards[board]
+        else:
+            raise Exception('The board is still not supported')
+
+    def get_part(self):
+        return self.part
+
+    def get_driver(self):
+        return self.driver
+
+    def get_board(self):
+        return self.board
+
+    def get_driver_path(self):
+        return '../templates/vivado_accelerator/' + self.board + '/' + self.driver + '_drivers/' + \
+               self.get_driver_file()
+
+    def get_driver_file(self):
+        driver_ext = '.py' if self.driver is 'python' else '.h'
```

Comment:
for some reason, I needed to change this line to 
```python
driver_ext = '.py' if self.driver == 'python' else '.h'
```
to get it to work for me. Otherwise, it was setting `driver_ext = '.h'` even though `self.driver = 'python'`